server/athenian/api/internal/settings.py

```diff
@@ -638,6 +638,27 @@ def repo_exists(self, logical_repo: RepositoryName) -> bool:
             return False
         return logical_repo.unprefixed in physical_repo_settings.logical_repositories
 
+    def explode_user_repos(self, repos: Iterable[str]) -> list[str]:
+        """Explode the list of repositories requested by user.
+
+        Each physical repo is exploded with its logical components if none of them is in the list.
+        Exploded physical repos are left in the result.
+        """
+        names = [RepositoryName.from_unprefixed(r) for r in repos]
+        referred_physical = {n.unprefixed_physical for n in names if n.is_logical}
+
+        exploded = []
+        for name in names:
```

Comment:
```suggestion
        for name in referred_physical:
```
because we don't have to do anything with physical repositories without logical occurrences. This explains why you catch `KeyError` below.